packages/core/src/components/forms/numericInput.tsx

```diff
@@ -201,9 +201,12 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
             "stepSize",
         ], true);
 
+        const isLarge = className != null && className.indexOf(Classes.LARGE) >= 0;
```

Comment:
I don't think this is so brittle... `active` and `loading` make a little more sense as props since you might change them over the lifecycle of the component... but the same is not true for `pt-large` or `pt-fill`. CSS classes work fine